src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,342 @@
+import logging
+import time
+from datetime import datetime, timedelta
+from typing import Any, TypedDict
+
+import sentry_sdk
+from django.conf import settings
+from google.api_core.exceptions import DeadlineExceeded, ServiceUnavailable
+from snuba_sdk import Column, Condition, Entity, Function, Op, Query, Request
+from snuba_sdk.orderby import Direction, OrderBy
+
+from sentry import features, nodestore
+from sentry.api.endpoints.group_similar_issues_embeddings import get_stacktrace_string
+from sentry.eventstore.models import Event
+from sentry.grouping.grouping_info import get_grouping_info
+from sentry.issues.grouptype import ErrorGroupType
+from sentry.issues.occurrence_consumer import EventLookupError
+from sentry.models.group import Group
+from sentry.models.grouphash import GroupHash
+from sentry.models.project import Project
+from sentry.seer.utils import (
+    CreateGroupingRecordData,
+    CreateGroupingRecordsRequest,
+    post_bulk_grouping_records,
+)
+from sentry.silo.base import SiloMode
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task
+from sentry.utils import json, metrics, redis
+from sentry.utils.iterators import chunked
+from sentry.utils.query import RangeQuerySetWrapper
+from sentry.utils.safe import get_path
+from sentry.utils.snuba import bulk_snuba_queries
+
+ITERATOR_CHUNK = 20
+SEER_BACKFILL_DELAY_PER_RECORD = 0.1
+BACKFILL_NAME = "backfill_grouping_records"
+LAST_PROCESSED_REDIS_KEY = "grouping_record_backfill.last_processed_id"
+
+logger = logging.getLogger(__name__)
+
+
+class GroupEventRow(TypedDict):
+    event_id: str
+    group_id: int
+    message: str
+
+
+class GroupStacktraceData(TypedDict):
+    data: list[CreateGroupingRecordData]
+    stacktrace_list: list[str]
+
+
+@instrumented_task(
+    name="sentry.tasks.backfill_seer_grouping_records",
+    queue="default",
+    max_retries=0,
+    silo_mode=SiloMode.REGION,
```

Comment:
let's set the `soft_time_limit` to 15 minutes and `time_limit` to 15 minutes and 5 seconds